streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java

```diff
@@ -711,19 +709,9 @@ public void shouldCloseOpenIteratorsWhenStoreIsClosedAndThrowInvalidStateStoreEx
         final WindowStoreIterator<String> iterator = windowStore.fetch(1, 1L, 3L);
         assertTrue(iterator.hasNext());
         windowStore.close();
-        try {
-            iterator.hasNext();
-            fail("should have thrown InvalidStateStoreException on closed store");
-        } catch (InvalidStateStoreException e) {
-            // ok
-        }
 
-        try {
```

Comment:
This was an expected condition before. This fix will suppress this Exception, do we want to consider another approach?

Reply:
Should this not still throw and `NoSuchElementException`? Not sure if we need to test for this here thought. It's just for clarification.

Reply:
`next()` does. I agree about the test maybe we should remove it?
WDYT?

\cc @guozhangwang @vvcephei

Reply:
Cannot follow with your comment. "maybe we should remove it" remove what?

Reply:
I think this test is fine -- make sense to ensure that `hasNext` does not throw. We should add a second test, that expects `next()` to throw IMHO.

Reply:
I agree we should keep the test.

The Iterator interface documents this relationship between `next()` and `hasNext()`. If it's in doubt whether `next()` will throw when `hasNext()` returns false, I think we should test it.